jerry-core/parser/js/js-parser-expr.c

```diff
@@ -353,6 +360,175 @@ parser_append_object_literal_item (parser_context_t *context_p, /**< context */
   }
 } /* parser_append_object_literal_item */
 
+#ifndef CONFIG_DISABLE_ES2015_CLASS
+/**
+ * Parse class as an object literal.
+ */
+void
+parser_parse_class_literal (parser_context_t *context_p, /**< context */
+                            lexer_literal_t *constructor_literal_p) /**< constructor literal */
+{
+  JERRY_ASSERT (context_p->token.type == LEXER_LEFT_BRACE);
+  JERRY_ASSERT (constructor_literal_p->type == LEXER_UNUSED_LITERAL);
+
+  parser_emit_cbc (context_p, CBC_CREATE_OBJECT);
+
+  parser_stack_push_uint8 (context_p, PARSER_OBJECT_PROPERTY_START);
+
+  bool is_static = false;
+
+  while (true)
+  {
+    if (!is_static)
+    {
+      lexer_expect_object_literal_id (context_p, false);
+    }
+    else
+    {
+      lexer_scan_identifier (context_p, true);
+
+      if (context_p->token.type == LEXER_LITERAL && context_p->token.lit_location.type == LEXER_IDENT_LITERAL)
+      {
+        lexer_construct_literal_object (context_p,
+                                        &context_p->token.lit_location,
+                                        LEXER_STRING_LITERAL);
+      }
```

Comment:
What are the other cases? Do `static static()` or `static get 1()` work? Cannot we merge the two cases?

Reply:
Both works and I've added the `static get 1` test case. Unfortunately we cannot merge these statements due to the `static` is a reserved keyword.